x-pack/plugins/alerting/server/task_runner/rule_loader.ts

```diff
@@ -110,6 +115,7 @@ export async function getRuleAttributes<Params extends RuleTypeParams>(
 
   const fakeRequest = getFakeKibanaRequest(context, spaceId, rawRule.attributes.apiKey);
   const rulesClient = context.getRulesClientWithRequest(fakeRequest);
+  const rulesSettingsClient = context.getRulesSettingsClientWithRequest(fakeRequest);
```

Comment:
I think there are a few problems with this.

1. Seems like an odd place to get the ruleSettingsClient.  Presumably it has to be done somewhere in the rule task runner, to get the current values to pass along internally.  If I was looking for it, I'd never look here though!  I think the TaskRunner constructor would be a better place.  I think `loadRule()` is used in other places as well, that probably don't need that client (just the SO attrs).

2. I don't think we want the fakeRequest here, since that's the user that created the rule.  They may not have privs to read the flapping config (as I understand it), and so presumably use of this client by such a user would end up throwing an error.  At least that's my general understanding of what's going on.  We need to do auth when users are reading/writing the settings in the UX (and HTTP APIs), but we will also need to get these settings for every rule run, and so the client for THOSE usages will need to be one of the superuser things.

Maybe since the settings aren't threaded all the way through yet, we could just remove these bits, since it doesn't appear they're doing anything.

Reply:
For # 2: Good point, In fact, this made me realize that I've completely neglected to use the `checkPrivileges` APIs to enforce `savedObject` privileges at the `rulesSettingsClient` level since I've been relying on the router `tag` property to gate APIs by features. So I will implement `checkPrivileges` shortly here as well. 

In that case, what is our preferred method of "faking" a superuser? Is there a `fakeRequest` as a superuser? Or should we just have a programmatic bypass of the privilege checks (seems kind of...gross, haha)?

Reply:
>In that case, what is our preferred method of "faking" a superuser? Is there a fakeRequest as a superuser? Or should we just have a programmatic bypass of the privilege checks (seems kind of...gross, haha)?

The recommended option is to create a scoped client that excludes the security extension. This will skip all the RBAC and pretty much uncouple a user from the SO client. You can find an example here: https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/rules_client_factory.ts#L94-L97

If that doesn't work and you can make the rules settings client work with a `savedObjectsClient` | `savedObjectsRepository`, you should be able by bypass RBAC whenever there isn't a user by creating a SO repository instead. The repository doesn't require a user and bypasses all the wrappers (encrypted saved objects, security, etc.) so it's less favourable but still works.